src/bucket.ts

```diff
@@ -3814,6 +3908,12 @@ class Bucket extends ServiceObject {
       options
     );
 
+    // Do not retry if precondition option ifMetagenerationMatch is not set
+    let numberOfRetries = this.storage.retryOptions.maxRetries;
```

Comment:
Do we want to name this `maxRetries` for consistency? Or is it confusing to use the same name again? I guess it's not hard to understand, but sometimes fewer than this number of retries will occur if it succeeds on the first retry (for example)